rust/src/parser.rs

```diff
@@ -26,6 +26,37 @@ use crate::applayer;
 use std::os::raw::{c_void,c_char,c_int};
 use crate::applayer::{AppLayerGetTxIterTuple};
 
+#[repr(C)]
+pub struct AppLayerReturn {
+    pub status: i32,
+    pub consumed: u32,
+    pub needed: u32,
+}
+
+impl AppLayerReturn {
+    pub fn ok() -> AppLayerReturn {
+        return AppLayerReturn {
+            status: 0,
+            consumed: 0,
+            needed: 0,
+        };
+    }
+    pub fn err() -> AppLayerReturn {
+        return AppLayerReturn {
+            status: -1,
+            consumed: 0,
+            needed: 0,
+        };
+    }
+    pub fn incomplete(consumed: u32, needed: u32) -> AppLayerReturn {
+        return AppLayerReturn {
+            status: 1,
+            consumed: consumed,
+            needed: needed,
+        };
+    }
+}
```

Comment:
So ok() means we consumed the exact number of bytes provided? It would essentially be the same as incomplete where consumed = input_len, and needed is 0?

Reply:
Essentially, yes.

Reply:
This might come up from others, so I'll bring it up here. I'm starting to like removing the `return` on simple methods like these.  But I am proposing to turn that lint off anyways, so it can state.

Also, this looks like a good candidate to be named AppLayerResult, but we don't quite follow the result pattern here, so I'd leave as is, maybe revisit separately later on to see if it can map to the Result pattern.